### PR TITLE
Proposal: Recommend providing TripUpdate predictions for the next trip in block

### PR DESCRIPTION
**Pull request by barbeau**
***Please do not merge***

@LeoFrachet @timMillet I wanted to get your feedback on this proposal prior to opening on the main google/transit repo. We discussed this a while back in our weekly meetings and everyone agreed it was a good idea, but I'd like comments on the exact language (e.g., "should" vs "must", should we include a time threshold like 1 hr?). Please let me know what you think.

Below is the full proposal for google/transit:

-------------
Previous discussions (1) (2) have raised the question of whether more than one TripUpdate should be included in a GTFS-realtime feed for a vehicle in the case where the vehicle is serving more than one trip in the same block. These discussions have concluded that multiple TripUpdates per vehicle are clearly beneficial in this case to avoid prediction "pop-in" for riders as the vehicle transitions from one trip to another and also give riders advance notice of delays that impact downstream trips (e.g., when the known delay exceeds planned layover times between trips).

However, this issue currently isn't addressed anywhere in the spec.

This proposal adds language to the spec recommending that multiple TripUpdates are included in a GTFS-realtime feed for vehicles running more than one trip in the same block.

1. GitHub PR proposal - "Define rules for propagating delays across trips in same block" - https://github.com/google/transit/pull/110
1. Google Groups post - "trip is in TripUpdates but not in VehiclePositions feed" - https://groups.google.com/forum/#!topic/gtfs-realtime/lVPOmi9A5vQ

### CODE REVIEW

**Comment by barbeau** [comment]
@timMillet Would you be able to review this in the next day or two?

**Comment by timMillet** [comment]
I'm ok with the proposal:
- The phrasing is clear and accurate.
- I prefer `Should` over `Must`, as it will always depend on the situation (e.g: it's ok if the producer does not provide trip updates for the next trip in the same block, if the next trip is in more than 1 hour).
- I would not include thresholds. The producer should provide trip updates as long as it is confident with its own data.

However, in reference.md, the subsequent paragraph is also describing a condition related to several trip updates for trips within the same block. This paragraph is also introducing the concept of block and referencing the GTFS Static spec. Have you thought about merging these 2 paragraphs (using bullet points to differentiate the conditions, maybe)?

Pro: Introduction of the block concept and the conditions related to it are located in a single paragraph.
Con: both paragraphs of this proposal are not the same anymore. 

Question on the side: I've always wondered why there is not just one file `reference.md` in the realtime spec (there are 4 files: `reference.md`, `trip-updates.md`, `vehicle-positions.md`, and `service-alerts.md`). Would you know what was the reason behind providing 3 extra files?

**Comment by barbeau** [comment]
>Have you thought about merging these 2 paragraphs (using bullet points to differentiate the conditions, maybe)?

Good idea! I just pushed a new commit that does this.

>Question on the side: I've always wondered why there is not just one file reference.md in the realtime spec (there are 4 files: reference.md, trip-updates.md, vehicle-positions.md, and service-alerts.md). Would you know what was the reason behind providing 3 extra files?

I think this mirrored the original Google GTFS-realtime website where there were introductory pages for each of the 3 main parts of the feed. It is awkward in a spec, though, where we end up having to duplicates things across pages. It's worth a discussion to see if we could combine them into a single doc.

File: gtfs-realtime/spec/en/reference.md [hunk 1]
```diff
@@ -133,9 +133,9 @@ Depending on the value of ScheduleRelationship, a TripUpdate can specify:
...
+If a vehicle is serving multiple trips within the same block:
+* the feed should include a TripUpdate for the trip currently being served by the vehicle as well as a another TripUpdate for the next trip in the block. Including multiple TripUpdates for the same vehicle avoids prediction "pop-in" for riders as the vehicle transitions from one trip to another and also gives riders advance notice of delays that impact downstream trips (e.g., when the known delay exceeds planned layover times between trips).
+* the respective TripUpdate entities are not required to be added to the feed in the same order that they are scheduled in the block (for more information about trips and blocks, please refer to [GTFS trips.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#tripstxt)). For example, if there are trips with `trip_ids` 1, 2, and 3 that all belong to one block, and the vehicle travels trip 1, then trip 2, and then trip 3, the `trip_update` entities may appear in any order - for example, adding trip 2, then trip 1, and then trip 3 is allowed.
```
Comment (timMillet) [inline gtfs-realtime/spec/en/reference.md]: `(for more information about trips and blocks, please refer to [GTFS trips.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#tripstxt))` should be moved from the second bullet point to the phrase introducing the 2 bullet points.

Comment (timMillet) [reply gtfs-realtime/spec/en/reference.md]: (from line 138 to line 136)

Comment (barbeau) [reply gtfs-realtime/spec/en/reference.md]: Good catch, thanks!

**Comment by barbeau** [comment]
Closing this draft pull request, as this proposal was merged upstream in the google/transit repo in https://github.com/google/transit/pull/206.

**Pull request closed without merging**